opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -240,7 +251,10 @@ def get_meter(
         return DefaultMeter()
 
 
-MetricT = TypeVar("MetricT", Counter, ValueRecorder, Observer)
+MetricT = TypeVar("MetricT", Counter, ValueRecorder)
+InstrumentT = TypeVar("InstrumentT", Counter, Observer, ValueRecorder)
+# TODO: Will populate with other observers when implemented
+ObserverT = TypeVar("ObserverT", DefaultObserver, ValueObserver)
```

Comment:
Using `TypeVar(..., bound=Observer)` here would do almost the same thing, right? Then we wouldn't need to update it as the rest of the instruments are added

Reply:
Yes you're right. Originally, I wanted `DefaultMeter` to only return `DefaultObserver` type on regeister_observer but mypy would complain about that (since it inherits from `Observer`). The api-sdk separation is really weird and introduces some strange architectural decisions.